carrier/carrier.go

```diff
@@ -102,11 +114,13 @@ func (evc *ethereumCarrier) SubscribeNewBlock(cb func(uint64), close chan bool)
 				close <- closed
 				return
 			case err := <-sub.Err():
-				if err := evc.reset(err); err.Error() != "tls: use of closed connection" {
-					zap.L().Fatal("failed to reset client", zap.Error(err))
-				}
-				if sub, err = evc.client.SubscribeNewHead(context.Background(), headers); err != nil {
-					zap.L().Fatal("failed to resubscribe new head", zap.Error(err))
+				for {
+					if sub, err = evc.retrySubscribeNewHead(err, headers); err == nil {
+						break
+					}
+					if err := evc.RotateClient(); err != nil {
```

Comment:
although we don't have lock in this struct, calling a public function is still something we want to prevent